core/domain/classifier_services.py

```diff
@@ -118,51 +95,90 @@ class StringClassifier(object):
 
     _DEFAULT_PREDICTION_THRESHOLD = 0.5
 
-    """Classifiers built with less than _DEFAULT_MIN_DOCS_TO_PREDICT will
-    likely not be useful for predicting as there are not enough examples to
-    build a generalized model. The value 20 was chosen as a balance between a
-    reasonable number of data to learn from and a low entry barrier to using
-    the classifier.
-    """
+    # Classifiers built with less than _DEFAULT_MIN_DOCS_TO_PREDICT will
+    # likely not be useful for predicting as there are not enough examples to
+    # build a generalized model. The value 20 was chosen as a balance between a
+    # reasonable number of data to learn from and a low entry barrier to using
+    # the classifier.
     _DEFAULT_MIN_DOCS_TO_PREDICT = 20
-    """Because prediction uses Prob(the doc should be assigned this label | the
-    doc is not assigned DEFAULT_LABEL), if there are only two labels (the
-    default label and one other) then the one other label will always be
-    predicted. Therefore, a minimum of 3 labels are needed to perform a valid
-    prediction.
-    """
+
+    # Because prediction uses Prob(the doc should be assigned this label | the
+    # doc is not assigned DEFAULT_LABEL), if there are only two labels (the
+    # default label and one other) then the one other label will always be
+    # predicted. Therefore, a minimum of 3 labels are needed to perform a valid
+    # prediction.
     _DEFAULT_MIN_LABELS_TO_PREDICT = 3
 
     DEFAULT_LABEL = '_default'
 
     def __init__(self):
         """Initializes constants for the classifier.
 
-        Setting a seed ensures that results are deterministic. There is nothing
-        special about the value 4. These should not be changed unless you know
-        what you're doing.
+        This class uses index notation with the format "_V_XY", where V is the
+        element of an array, and X and Y are the indices used to measure V.
+        https://en.wikipedia.org/wiki/Index_notation
+
+        The following maps index notation letters to their meanings:
+            b - boolean value for whether Y is set in X
+            c - count of Y's in X
+            p - position of a value V in X
+            l - label id
+            w - word id
+            d - doc id
         """
+
+        # Setting a seed ensures that results are deterministic.
+        # There is nothing special about the value 4.
         numpy.random.seed(seed=4)
 
+        # Internal learning rates. These are initialized to Wikipedia's
```

Comment:
Actually, I think the "These are initialized..." part should probably go with _DEFAULT_ALPHA and _DEFAULT_BETA, not here. That's the place where people are most likely to change the constants.